### PR TITLE
Added tests of WebAudioPlayer APIs to the client component tests

### PR DESCRIPTION
**Pull request by skyshaha1**
Summary: Test the WebAudioPlayer APIs in the client component tests - CreateWebAudioPlayerTest.
Type: Feature
Test Plan: Unittests, Component Tests.
Jira: RIALTO-415

### CODE REVIEW

File: tests/componenttests/client/stubs/WebAudioPlayerModuleStub.cpp [hunk 1]
```diff
@@ -18,11 +18,37 @@
...
+#include <IIpcServer.h>
+#include <IIpcServerFactory.h>
 #include <gtest/gtest.h>
 #include <memory>
 #include <string>
 #include <vector>
 
+namespace
+{
+firebolt::rialto::WebAudioPlayerStateEvent_WebAudioPlayerState
```
Comment (skywilliamsonl) [inline tests/componenttests/client/stubs/WebAudioPlayerModuleStub.cpp:30]: Make a common header file with this conversion for both unittests and component tests, like we do in MediaPipeline. This should be used by WebAudioPlayerModuleServiceTestsFixture.cpp too.

File: tests/componenttests/client/tests/base/WebAudioPlayerTestMethods.cpp [hunk 2]
```diff
@@ -28,6 +28,9 @@ namespace
 const std::string kAudioMimeType{"audio/x-raw"};
 const uint32_t kPriority{5};
 const int32_t kWebAudioPlayerHandle{1};
+constexpr uint32_t kPreferredFrames{1};
+constexpr uint32_t kMaximumFrames{0};
```
Comment (skywilliamsonl) [inline tests/componenttests/client/tests/base/WebAudioPlayerTestMethods.cpp:32]: Max frames should be none zero

**Review by aczs — APPROVED** [review]

File: tests/componenttests/client/tests/webaudio/CreateWebAudioPlayerTest.cpp [hunk 3]
```diff
@@ -54,7 +54,16 @@ class CreateWebAudioPlayerTest : public ClientComponentTest
  *   Expect that  web audio api is called on the server
  *   Check that the object returned is valid.
```
Comment (skywilliamsonl) [inline tests/componenttests/client/tests/webaudio/CreateWebAudioPlayerTest.cpp:55]: Need to add the client check here, like we do for MediaPipeline test

**Review by skywilliamsonl — COMMENTED** [review]
Still an outstanding comment in the previous PR about using common matcher in unittests.

File: tests/componenttests/client/tests/webaudio/CreateWebAudioPlayerTest.cpp (hunk 3 above)
Comment (skywilliamsonl) [reply tests/componenttests/client/tests/webaudio/CreateWebAudioPlayerTest.cpp:55]: This check is for web audio player client, and you also need to add a new test method to check this, same as we do for media pipeline MediaPipelineTestMethods::checkMediaPipelineClient();

File: tests/unittests/media/server/ipc/webAudioPlayerModuleService/WebAudioPlayerModuleServiceTestsFixture.cpp [hunk 4]
```diff
@@ -18,6 +18,7 @@
  */
 
 #include "WebAudioPlayerModuleServiceTestsFixture.h"
+#include "MediaPipelineProtoUtils.h"
```
Comment (skywilliamsonl) [inline tests/unittests/media/server/ipc/webAudioPlayerModuleService/WebAudioPlayerModuleServiceTestsFixture.cpp:21]: Should be WebAudio & you need to delete the convertWebAudioPlayerState that's also defined in here.

File: tests/componenttests/client/tests/base/WebAudioPlayerTestMethods.cpp [hunk 5]
```diff
@@ -74,6 +82,38 @@ void WebAudioPlayerTestMethods::destroyWebAudioPlayer()
...
+            DoAll(SetArgPointee<2>(m_webAudioPlayerModuleMock->webAudioGetDeviceInfoResponse(kPreferredFrames,
+                                                                                             kMaximumFrames,
+                                                                                             kSupportDeferredPlay)),
+                  (WithArgs<0, 3>(Invoke(&(*m_webAudioPlayerModuleMock), &WebAudioPlayerModuleMock::defaultReturn)))));
+}
+void WebAudioPlayerTestMethods::getDeviceInfo()
+{
+    uint32_t preferredFrames = 0;
+    uint32_t maximumFrames = 0;
+    bool supportDeferredPlay = true;
```
Comment (skywilliamsonl) [inline tests/componenttests/client/tests/base/WebAudioPlayerTestMethods.cpp:109]: This should be default false.

**Review by skywilliamsonl — APPROVED** [review]

**Review by aczs — APPROVED** [review]

**Pull request merged**